tests/test_main.cpp

```diff
@@ -147,6 +147,7 @@
 #include "tests/servers/test_navigation_server_3d.h"
 #endif // MODULE_NAVIGATION_ENABLED
 
+#include "tests/core/io/test_json_native.h"
```

Comment:
Why is this in the `#ifndef _3D_DISABLED` block?

It should be further up next to all other `tests/core/io/` tests.

Reply:
See above, it uses 3D code

Reply:
Then only the 3D code should be ifdef'ed, we should still unit test this feature even if 3D is disabled.
I also don't think it has any reason to depend on 3D in the first place, at least for a first basic unit test that doesn't aim to cover all kinds of node conversions.